nvflare/app_common/utils/fl_model_utils.py

```diff
@@ -104,36 +103,42 @@ def from_shareable(
         In the future, we should be using the to_dxo, from_dxo directly.
         And all the components should be changed to accept the standard DXO.
         """
-        kwargs = {}
         dxo = from_shareable(shareable)
+        metrics = None
+        params_type = None
+        params = None
 
         if dxo.data_kind == DataKind.METRICS:
-            kwargs[FLModelConst.METRICS] = dxo.data
+            metrics = dxo.data
         else:
             params_type = data_kind_to_params_type.get(dxo.data_kind)
             if params_type is None:
                 raise ValueError(f"Invalid shareable with dxo that has data kind: {dxo.data_kind}")
-            kwargs[FLModelConst.PARAMS_TYPE] = ParamsType(params_type)
+            params_type = ParamsType(params_type)
             if params_converter:
                 dxo.data = params_converter.convert(dxo.data)
-            kwargs[FLModelConst.PARAMS] = dxo.data
+            params = dxo.data
 
         current_round = shareable.get_header(AppConstants.CURRENT_ROUND, None)
         total_rounds = shareable.get_header(AppConstants.NUM_ROUNDS, None)
         validate_type = shareable.get_header(AppConstants.VALIDATE_TYPE, None)
 
-        kwargs[FLModelConst.CURRENT_ROUND] = current_round
-        kwargs[FLModelConst.META] = dxo.meta
+        meta = dict(dxo.meta)
         if validate_type is not None:
-            kwargs[FLModelConst.META][MetaKey.VALIDATE_TYPE] = validate_type
-        if total_rounds is not None:
-            kwargs[FLModelConst.TOTAL_ROUNDS] = total_rounds
+            meta[MetaKey.VALIDATE_TYPE] = validate_type
 
         if fl_ctx is not None:
-            kwargs[FLModelConst.META][MetaKey.JOB_ID] = fl_ctx.get_job_id()
-            kwargs[FLModelConst.META][MetaKey.SITE_NAME] = fl_ctx.get_identity_name()
-
-        result = FLModel(**kwargs)
+            meta[MetaKey.JOB_ID] = fl_ctx.get_job_id()
+            meta[MetaKey.SITE_NAME] = fl_ctx.get_identity_name()
+
+        result = FLModel(
+            params_type=params_type,
+            params=params,
+            metrics=metrics,
+            current_round=current_round,
+            total_rounds=total_rounds,
+            meta=meta,
```

Comment:
Should we treat the current_round and total_rounds part of the "meta"?

Reply:
I am neutral on this, @holgerroth @chesterxgchen what do you think?